### PR TITLE
Refactor and clean up harvester

### PR DESCRIPTION
**Pull request by ruflin**

### CODE REVIEW

File: crawler/registrar.go [hunk 1]
```diff
@@ -159,6 +162,7 @@ func (r *Registrar) fetchState(filePath string, fileInfo os.FileInfo) (int64, bo
 		// We're resuming - throw the last state back downstream so we resave it
 		// And return the offset - also force harvest in case the file is old and we're about to skip it
 		r.Persist <- lastState
+		logp.Debug("Existing file with offset: %s, offset: %s", filePath, lastState.Offset)
```
Comment (tsg) [inline crawler/registrar.go]: missing module name here.

File: crawler/registrar.go [hunk 2]
```diff
@@ -197,8 +201,8 @@ func (r *Registrar) getPreviousFile(newFilePath string, newFileInfo os.FileInfo)
 
 		// Compare states
 		if newState.IsSame(oldState.FileStateOS) {
+			logp.Debug("Old file with new name found: %s is no %s", oldFilePath, newFilePath)
```
Comment (tsg) [inline crawler/registrar.go]: missing module name here.

**Comment by ruflin** [comment]
@tsg Thanks for spotting these. I fixed the issue and added a few more improvements in https://github.com/elastic/filebeat/commit/542b4ad56406629e1fbb42d34a083447dda39c31

File: crawler/registrar.go [hunk 3]
```diff
@@ -91,12 +91,12 @@ func (r *Registrar) Run() {
 		var events []*FileEvent
 		select {
 		case <-r.done:
-			logp.Debug("registrar", "Ending Registrar")
+			logp.Info("Ending Registrar")
```
Comment (urso) [inline crawler/registrar.go]: maybe have this message in defer statement?

**Comment by urso** [comment]
LGTM

File: crawler/registrar.go (hunk 3 above)
Comment (ruflin) [inline crawler/registrar.go]: Probably yes as soon as we have one. I would keep it there for the moment.

**Pull request merged**